src/models/github-server-app.ts

```diff
@@ -30,6 +30,23 @@ export class GitHubServerApp extends Model {
 	updatedAt: Date;
 	createdAt: Date;
 
+	getEncryptionSecretKey() {
+		return EncryptionSecretKeyEnum.GITHUB_SERVER_APP;
+	}
+
+	async getEncryptContext() {
+		//For example: we can call database to fetch sharedSecret for use as EncryptionContext
```

Comment:
I am hesitate to put it here right now, as this is under discussion.

Reply:
Well, we should probably decide this before merging it before someone uses the model and we can potentially lose data if we don't migrate appropriately.

Reply:
I thought about that, yup we can do this way. But I suspect in reality we will sooner or later encounter a situation that we need to change the encryptionContext.

I suggest if that happen, we can change that `async getEncryptContext()` to return `EncryptionContext[]` instead of `EncryptionContext`. And in the base class, it decrypt by the order of the context in the array.

Reply:
hm, not a bad solution, we would need to re-encrypt after decryption to work with the "main" context or figure out a way to do this via sequelize migrations as a once off...